Reworked metrics collection.

ydrozhdzhal opened the pull request.

mwear committed:
statsdreceiver: fix start timestamp / temporality for counters (#5714)

* statsdreceiver: fix start timestamp / temporality for counters

* Update receiver/statsdreceiver/protocol/metric_translator.go

Co-authored-by: Joshua MacDonald <jmacd@users.noreply.github.com>

Co-authored-by: Joshua MacDonald <jmacd@users.noreply.github.com>

rmfitzpatrick committed:
Better document SignalFx access token passthrough feature (#5165)

bogdandrutu committed:
Move sharedcomponent to its own package (#5832)

The goal is to remove from coreinternal everything that depends on the core (and not only on model), to
ensure that packages like the translators do not depend on anything more than just model.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

Commit:
Migrated Google Cloud Spanner receiver to scraper approach.

dmitryax committed:
Update configuration default values in log receivers docs (#5840)

Update default values in `filelog`, `tcplog` and `udplog` receivers according to the recent changes in the log-collection library:
- https://github.com/open-telemetry/opentelemetry-log-collection/pull/147
- https://github.com/open-telemetry/opentelemetry-log-collection/pull/261

NathanielRN committed:
rpc.system has priority to determine aws namespace in awsxrayexporter (#5833)

asukhyy commented on receiver/googlecloudspannerreceiver/internal/metadata/metricsdatapoint.go:
```diff
@@ -0,0 +1,77 @@
+
+type MetricsDataPointGroupingKey struct {
+	MetricName     string
+	Timestamp      time.Time
```
Didn't get why `Timestamp` is needed here it looks to me that it is wrong.

ydrozhdzhal replied:
I am waiting for better proposal from your side.

ydrozhdzhal replied:
Reworked grouping conception. Timestamp is no longer part of grouping key.

mx-psi committed:
[exporter/datadogexporter] Remove spammy logging (#5856)

KSerrania committed:
[datadogexporter] Remove obsolete report_buckets config (#5858)

* [datadogexporter] Remove obsolete report_buckets config

* Add comments to config unmarshall

bogdandrutu committed:
Promote Alex Boten to maintainer (#5865)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

vikrambe committed:
Add support for composite sampling policy to the tailsampler (#4958)

Is your feature request related to a problem? Please describe.
#1410
Design Doc - Added support for composite policy in tailsampling processor. This would help in grouping sampling policies and rate limiting them. https://docs.google.com/document/d/10wpIv3TtXgOik05smHm3nYeBX48Bj76TCMxPy8e1NZw/edit#heading=h.ecy5l2puwtp4
This is a split. Refer PR open-telemetry/opentelemetry-collector#1894 (comment)

Due to EasyCLA issue opening a new PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/4396

Link to tracking Issue: 1306

tigrannajaryan committed:
Improve performance of metric expression matcher (#5864)

Refactored env struct to avoid creating closure functions for every
expression evaluation operation. This reduced the allocations and
increased the speed.

Improvements:

$ benchstat old.txt new.txt
name          old time/op    new time/op    delta
ExprFilter-8     537µs ± 2%     476µs ± 1%  -11.33%  (p=0.000 n=9+10)

name          old alloc/op   new alloc/op   delta
ExprFilter-8     151kB ± 0%     102kB ± 0%  -32.43%  (p=0.000 n=8+8)

name          old allocs/op  new allocs/op  delta
ExprFilter-8     8.44k ± 0%     6.39k ± 0%  -24.24%  (p=0.000 n=10+10)

bogdandrutu committed:
[deltatocomulativeprocessor]: Move tracking to internal (#5871)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
[cumulativetodeltaprocessor]: small nit, avoid creating object until is necessary (#5872)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>
